java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/common/MySqlValidator.java

```diff
@@ -180,6 +180,7 @@ private void validatePrivileges() throws SQLException {
                 String granted = res.getString(1).toUpperCase();
                 // all privileges granted, check passed
                 if (granted.contains("ALL")) {
+                    hashSet.clear();
```

Comment:
I checked that the `ALL` doesn't contains `REPLICATION SLAVE` and `REPLICATION CLIENT`, so would you mind just remove the lines 182-184?

Reply:
Sorry, I'm confused why the `ALL` doesn't contains `REPLICATION SLAVE` and `REPLICATION CLIENT`. I checked the mysql development documentation, `ALL` means “all privileges available at a given privilege level”. And I executing the following SQL statement, there is no prompt indicating that there is 'Access denied'.

Check grants
```sql
SHOW GRANTS FOR CURRENT_USER(); 
# GRANT ALL PRIVILEGES ON *.* TO 'root'@'%' WITH GRANT OPTION
```
Executes SQL that can only be executed with `REPLICATION SLAVE` and `REPLICATION CLIENT`
```sql
SHOW MASTER STATUS;
SHOW BINLOG EVENTS;
```

see
- https://dev.mysql.com/doc/refman/8.0/en/privileges-provided.html#priv_all
- https://dev.mysql.com/doc/refman/8.0/en/privileges-provided.html#priv_replication-client
- https://dev.mysql.com/doc/refman/8.0/en/privileges-provided.html#priv_replication-slave

Reply:
I think becasue you are using the `root` user, which by default have the `REPLICATION SLAVE` and `REPLICATION CLIENT`.

I tested mysql 8.0 with a new created user bob and grant all to the `bob`.
```
mysql> show grants;
+-----------------------------------------------+
| Grants for bob@%                              |
+-----------------------------------------------+
| GRANT USAGE ON *.* TO `bob`@`%`               |
| GRANT ALL PRIVILEGES ON `mydb`.* TO `bob`@`%` |
+-----------------------------------------------+
2 rows in set (0.00 sec)

mysql> show master status;
ERROR 1227 (42000): Access denied; you need (at least one of) the SUPER, REPLICATION CLIENT privilege(s) for this operation
```
May I know which version of mysql you are using? The show grants of of my mysql 8.0 list all the privileges explicitly instead of just a single line `GRANT ALL PRIVILEGES ON *.* TO 'root'@'%' WITH GRANT OPTION`.

Reply:
It seems to be because of the mysql version.

```shell
mysql> SELECT VERSION();
+------------+
| VERSION()  |
+------------+
| 5.7.36-log |
+------------+
```
I also tested mysql 5.7 with a new created user bob and grant all to the bob.

```shell
mysql> CREATE USER 'bob'@'%' IDENTIFIED BY 'password';
Query OK, 0 rows affected (0.00 sec)

mysql> GRANT ALL PRIVILEGES ON *.* TO 'bob'@'%';
Query OK, 0 rows affected (0.01 sec)

mysql> FLUSH PRIVILEGES;
Query OK, 0 rows affected (0.01 sec)

mysql> SHOW GRANTS FOR 'bob';
+------------------------------------------+
| Grants for bob@%                         |
+------------------------------------------+
| GRANT ALL PRIVILEGES ON *.* TO 'bob'@'%' |
+------------------------------------------+
1 row in set (0.00 sec)

mysql -h 10.8.16.9 -P 3306 -u bob -ppassword

mysql> show grants;
+------------------------------------------+
| Grants for bob@%                         |
+------------------------------------------+
| GRANT ALL PRIVILEGES ON *.* TO 'bob'@'%' |
+------------------------------------------+

mysql> SHOW MASTER STATUS;
+------------------+----------+--------------+------------------+-------------------+
| File             | Position | Binlog_Do_DB | Binlog_Ignore_DB | Executed_Gtid_Set |
+------------------+----------+--------------+------------------+-------------------+
| mysql-bin.000044 |     4818 |              |                  |                   |
+------------------+----------+--------------+------------------+-------------------+
1 row in set (0.00 sec)
```

So maybe consider compatibility with mysql 5.7?

Reply:
Thanks for your feedback, I'll fix the issue https://github.com/risingwavelabs/risingwave/issues/12501